### ISSUE TITLE
TypeScript unhappy with 'blockly/javascript' exports

### ISSUE THREAD

**Issue by thepiper** [google/blockly#7283]
## TL;DR 

```ts
import { Order } from 'blockly/javascript';
```
Fails to compile without `// @ts-ignore` flag

### Temporary fix

Adding `// @ts-ignore` comment will allow it to compile and run without any noted errors
```ts
// @ts-ignore
import { Order } from 'blockly/javascript';
```

### Details

Didn't have time to look too deeply into the issue, but this is what I've dug up from a cursory look:

Looks like `Order` is not being exported in the `javascript.d.ts` file - https://github.com/google/blockly/blob/blockly-v10.0.1/typings/javascript.d.ts

Might be because typescript does not like the `export * from ...` syntax in original `javascript.js` source - https://github.com/google/blockly/blob/blockly-v10.0.1/generators/javascript.js

**Comment by cpcallen** [comment]
Yes: the `.d.ts` files did not get updated, and they should have.

I think this slipped my mind because the intention was (and remains) to get these fully converted to TypeScript and actually _generate_ the `.d.ts` files rather than maintain them manually.

(Aside: same for library blocks, where I think we may be overwriting the generated `.d.ts` files with the manually-curated ones before publication.)

**Comment by pavi2410** [comment]
I'm still unable to import the `Order` enum with Blockly v10.2.2 on NodeJS 21

```ts
import { javascriptGenerator, Order } from "blockly/javascript";
                              ^
SyntaxError: The requested module 'blockly/javascript' does not provide an export named 'Order'
```
